alpha/hal.cpp

```diff
@@ -324,7 +328,47 @@ void _TS_HAL::log_init()
 #endif
 }
 
-
+//
+// WIFI setup
+// 
+void _TS_HAL::wifi_init()
+{
+  if (!this->wifiInitialized)
+  {
+  	WiFi.mode(WIFI_STA);
+  	WiFi.disconnect();
+  	int n = 0;
+  	n = WiFi.scanNetworks();
+  	if (n > 3)
+  	{
+  		n = 3;
+  	}
+  	for (int i = 0; i < n; ++i)
+  	{
+  		//prints top 3 nearest WIFI SSIDS
+  		log(WiFi.SSID(i));
+  		delay(1000);
+  	}
+  
+  	if (n == 0)
+  	{
+  		log("no wifi found \n");
+  	}
+  	else
+  	{
+  		//Connect to, prints name of connected WIFI. 
+  		WiFi.begin(WIFI_SSID, WIFI_PASS);
+      while (WiFi.status() != WL_CONNECTED) 
```

Comment:
no while loops/delays here.

Reply:
convert this whole logic into smaller functions to allow polling, or use event callbacks if they exist.